flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/DefaultScheduler.java

```diff
@@ -324,13 +332,24 @@ private void restartTasksWithDelay(final FailureHandlingResult failureHandlingRe
                                 .values());
         final boolean globalRecovery = failureHandlingResult.isGlobalFailure();
 
+        if (globalRecovery) {
```

Comment:
It's highly related to the change of `SpeculativeScheduler` (otherwise we do not need to do it) to avoid printing the `restarting` log when just trying to create a `failureHandlingResult`. Therefore I think they are valid to be in the same commit.